### DISCUSSION TITLE
Dynamic routes

### DISCUSSION THREAD

**Discussion by rinkusam12** [vercel/next.js#24678] [Help]
Is it possible to do something like this?

I have one dynamic page with [slug].js and one with [...slug].js and I am using getStaticProps with fallback true.
What I want is if the [slug].js API returns no data for a particular route then it should pass the control to [...slug].js.
For example if /flower doesn't provide any data in [slug].js page then check if data is coming in [...slug].js and render page accordingly.

Currently what is happening is if the data doesn't come in [...slug].js then it returns a 404 page.

**Comment by bobarros** [comment]
What you want was suppose to be the regular behavior:

From documents: 

_Predefined routes take precedence over dynamic routes, and dynamic routes over catch all routes. Take a look at the following examples:_

_pages/post/create.js - Will match /post/create_
_pages/post/[pid].js - Will match /post/1, /post/abc, etc. But not /post/create_
_pages/post/[...slug].js - Will match /post/1/2, /post/a/b/c, etc. But not /post/create, /post/abc_

If you remove [...slug].js, does your [slug].js takes control and return anything?
